WalletWasabi.Fluent/CrashReport/CrashReporter.cs

```diff
@@ -16,22 +20,36 @@ public static void Invoke(Exception exceptionToReport)
 				var serializedException = exceptionToReport.ToSerializableException();
 				var base64ExceptionString = SerializableException.ToBase64String(serializedException);
 				var args = $"crashreport -exception=\"{base64ExceptionString}\"";
-
-				var path = Process.GetCurrentProcess().MainModule?.FileName;
-				if (string.IsNullOrEmpty(path))
-				{
-					throw new InvalidOperationException($"Invalid path: '{path}'");
-				}
-
-				ProcessStartInfo startInfo = ProcessStartInfoFactory.Make(path, args);
-				using Process? p = Process.Start(startInfo);
+				StartWasabiWithArgs(args);
 			}
 			catch (Exception ex)
 			{
 				Logger.LogWarning($"There was a problem while invoking crash report: '{ex}'.");
 			}
 		}
 
+		public static void RestartWasabi()
+		{
+			Task.Run(async () =>
```

Comment:
I think the order will always be start the new process first, wait 1s then shutdown the crashreporter.. they arent dependent on one another since the process will fork anyway

Reply:
I agree with David on it is time-sensitive... I mean I am sure it is working on 99% of the machines, but there is the 1% super extra mega giga powerful machines... where the scenario would be:
 
 1. Request Shutdown on the first instance (wait 1 sec).
 2. Start Wasabi - second instance 
 3. Wasabi has started, singe instance checker detects the first instance -> Error
 4. 1 sec elapsed, first instance shutdown